From 94b7f26aeda31cf55ff9b7c04f390ba6d35977e4 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Thu, 31 Oct 2013 08:57:56 -0700
Subject: [PATCH 2/2] miext/damage: Partial revert of "Only wrap into the GC
 ops chain if there's a listener (v3)"

Fixes regression from: 4dc2a76740d921c824a4d8193f39dd373475f02a

http://lists.x.org/archives/xorg-devel/2013-May/036241.html
https://trac.macports.org/ticket/38993

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Reviewed-by: Adam Jackson <ajax@redhat.com>
Signed-off-by: Keith Packard <keithp@keithp.com>
(cherry picked from commit abf5d5ac12437ebe156b4dd500c2acd69eea3654)
---
 miext/damage/damage.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git xorg-server-1.14.4/miext/damage/damage.c xorg-server-1.14.4/miext/damage/damage.c
index a98c20e..3dc3180 100644
--- xorg-server-1.14.4/miext/damage/damage.c
+++ xorg-server-1.14.4/miext/damage/damage.c
@@ -439,10 +439,7 @@ damageValidateGC(GCPtr pGC, unsigned long changes, DrawablePtr pDrawable)
     drawableDamage(pDrawable);
     DAMAGE_GC_FUNC_PROLOGUE(pGC);
     (*pGC->funcs->ValidateGC) (pGC, changes, pDrawable);
-    if (pDamage)
-        pGCPriv->ops = pGC->ops; /* so it's not NULL, so FUNC_EPILOGUE does work */
-    else
-        pGCPriv->ops = NULL;
+    pGCPriv->ops = pGC->ops; /* just so it's not NULL */
     DAMAGE_GC_FUNC_EPILOGUE(pGC);
 }
 
-- 
1.8.4.1

